pygluu/kubernetes/gui.py

```diff
@@ -0,0 +1,290 @@
+def agreement():
+    """Input for Accepting license
+    """
+
+    if request.method == 'POST':
+        next_step = request.form['next_step']
+        default_settings['ACCEPT_GLUU_LICENSE'] = request.form['accept_gluu_license']
+        return redirect(url_for(next_step))
+
+    with open('./LICENSE', 'r') as f:
+        agreement_file = f.read()
+
+    return render_template('index.html',
+                           license=agreement_file,
+                           step='license',
+                           next_step='gluu_version')
+
+
+@app.route('/gluu_version', methods=['GET', 'POST'])
+def gluu_version():
+    """Input for Gluu versions
+    """
+
+    versions, version_number = get_supported_versions()
+
+    if request.method == 'POST':
+        next_step = request.form['next_step']
+        default_settings['GLUU_VERSION'] = request.form['gluu_version']
+        image_names_and_tags = versions.get(default_settings['GLUU_VERSION'], {})
+        default_settings.update(image_names_and_tags)
+
+        return redirect(url_for(next_step))
+
+    supported_versions = []
+    for k, v in versions.items():
+        if "_dev" in k:
+            supported_versions.append((version_number, k))
+        else:
+            if float(k) > version_number:
+                supported_versions.append((version_number, float(k)))
+
+    return render_template('index.html',
+                           supported_versions=supported_versions,
+                           step='gluu_version',
+                           next_step='deployment_arch')
+
+
+@app.route('/deployment-arch', methods=['GET', 'POST'])
+def deployment_arch():
+    """
+    Input for the kubernetes infrastructure used.
+    """
+    if request.method == 'POST':
+        next_step = request.form['next_step']
+        default_settings['DEPLOYMENT_ARCH'] = request.form['deployment_arch']
+        return redirect(url_for(next_step))
+
+    return render_template('index.html',
+                           step='deployment_arch',
+                           next_step='gluu_namespace')
+
+
+@app.route('/gluu-namespace', methods=['GET', 'POST'])
+def gluu_namespace():
+    """
+    Input for gluu namespace.
+    """
+    if request.method == 'POST':
+        next_step = request.form['next_step']
+        default_settings['GLUU_NAMESPACE'] = request.form['gluu_namespace']
+        return redirect(url_for(next_step))
+
+    return render_template('index.html',
+                           step='gluu_namespace',
+                           next_step='optional_services')
+
+
+@app.route('/optional-services', methods=['GET', 'POST'])
+def optional_services():
+    """
+    Input for optional services.
+    """
+    if request.method == 'POST':
+        for i in request.form.keys():
+            if i == 'next_step':
+                next_step = request.form[i]
+                continue
+            default_settings[i.upper()] = request.form[i]
+
+        return redirect(url_for(next_step))
```

Comment:
`next_step` is defined in if clause. This may bring up a reference before assignment error